controllers/argocd/ingress.go

```diff
@@ -162,7 +162,11 @@ func (r *ReconcileArgoCD) reconcileArgoServerIngress(cr *argoprojv1a1.ArgoCD) er
 	if err := controllerutil.SetControllerReference(cr, ingress, r.Scheme); err != nil {
 		return err
 	}
-	return r.Client.Create(context.TODO(), ingress)
+
+	if !found {
+		return r.Client.Create(context.TODO(), ingress)
+	}
+	return r.Client.Update(context.TODO(), ingress)
```

Comment:
Hi @mosuke5 Thanks for your PR! 
Can we add a check here to see if anything has changed before calling Update? Otherwise we might have a lot of unnecessary api calls being made

Reply:
@jaideepr97 Thank you for your review. I fixed to update only when  annotations and spec of ingress resource are changed.